zcash_test_vectors/transaction_zsa.py

```diff
@@ -0,0 +1,157 @@
+        self.rseed = rand.b(32)
+
+    def __bytes__(self):
+        ret = b''
+        ret += bytes(self.recipient)
+        ret += struct.pack('<Q', self.value)
+        ret += bytes(self.assetBase)
+        ret += bytes(self.rho)
+        ret += self.rseed
+
+        return ret
+
+
+class TransactionZSA(TransactionBase):
+    def __init__(self, rand, consensus_branch_id, have_orchard_zsa = True, have_burn = True, have_issuance = True):
+
+        # Since burn is part of the OrchardZSA bundle, ensure that there are no burn fields
+        # when there are no OrchardZSA fields
+        assert have_orchard_zsa or not have_burn
+
+        # All the Transparent and Sapling Transaction Fields are initialized in the super (TransactionBase) class.
+        super().__init__(rand, have_orchard_zsa)
+
+        # Common Transaction Fields
+        self.nVersionGroupId = NU7_VERSION_GROUP_ID
+        self.nConsensusBranchId = consensus_branch_id
+
+        # Orchard-ZSA Transaction Fields
+        if have_orchard_zsa:
+            for _ in range(rand.u8() % 5):
+                self.vActionsOrchard.append(OrchardZSAActionDescription(rand))
+            self.flagsOrchard = (self.flagsOrchard & 7) | 4  # Only three flag bits are currently defined, and we set enableZSA to true.
+            if self.is_coinbase():
+                # set enableSpendsOrchard = 0
+                self.flagsOrchard &= 2
+
+        # OrchardZSA Burn Fields
+        self.vAssetBurnOrchardZSA = []
+        if have_burn:
+            for _ in range(rand.u8() % 5):
+                self.vAssetBurnOrchardZSA.append(AssetBurnDescription(rand))
+
+        # ZSA Issuance Fields
+        self.vIssueActions = []
+        if have_issuance:
+            self.ik = IssuanceKeys(rand.b(32)).ik
+            for _ in range(rand.u8() % 5):
+                self.vIssueActions.append(IssueActionDescription(rand, self.ik))
+            self.issueAuthSig = rand.b(64)
+
+    def version_bytes(self):
+        return NU7_TX_VERSION | (1 << 31)
+
+    def orchard_zsa_burn_field_bytes(self):
+        ret = b''
+        ret += write_compact_size(len(self.vAssetBurnOrchardZSA))
+        if len(self.vAssetBurnOrchardZSA) > 0:
+            for desc in self.vAssetBurnOrchardZSA:
+                ret += bytes(desc)
+        return ret
+
+    def issuance_field_bytes(self):
+        ret = b''
+        ret += write_compact_size(len(self.vIssueActions))
+        if len(self.vIssueActions) > 0:
+            for desc in self.vIssueActions:
+                ret += bytes(desc)
+            ret += self.ik
+            ret += bytes(self.issueAuthSig)
+        return ret
+
+    def __bytes__(self):
+        ret = b''
+
+        # Fields that are in TransactionBase: Common, Transparent, Sapling, most Orchard
+        ret += super().__bytes__(self.version_bytes(), self.nVersionGroupId, self.nConsensusBranchId)
+
+        # OrchardZSA remaining Transaction Fields (if the Orchard bundle exists)
```

Comment:
`# OrchardZSA transaction fields` (x2)

Reply:
This feels like it could be confusing. Only a few of the OrchardZSA fields come under this comment, the rest go under the "common" section since they fit inside the TransactionBase

Reply:
I still prefer the shorter suggestions, we have a "common" part and an "additional specific" part.

Aside, the words "reaming"(remaining from what? the reader should keep a context for comments) and "that are in" (a little bit like story telling) are not the best fit. Should be short and concise..

Reply:
Okay, done.